rero_ils/modules/libraries/api.py

```diff
@@ -221,47 +142,101 @@ def _has_is_open(self):
                     return True
         return False
 
+    def _get_exceptions_matching_date(self, date_to_check, day_only=False):
+        """Get all exception matching a given date."""
+        for exception in self.get('exception_dates', []):
+            # Get the start date and the gap (in days) between start date and
+            # end date. If no end_date are supplied, the gap will be 0.
+            start_date = date_string_to_utc(exception['start_date'])
+            end_date = start_date
+            day_gap = 0
+            if exception.get('end_date'):
+                end_date = date_string_to_utc(exception.get('end_date'))
+                day_gap = (end_date - start_date).days
+
+            # If the exception is repeatable, then the start_date should be the
+            # nearest date (lower or equal than date_to_check) related to the
+            # repeat period/interval definition. To know that, we need to know
+            # all exception dates possible (form exception start_date to
+            # date_to_check) and get only the last one.
+            if exception.get('repeat'):
+                period = exception['repeat']['period'].upper()
+                exception_dates = rrule(
+                    freq=FREQNAMES.index(period),
+                    until=date_to_check,
+                    interval=exception['repeat']['interval'],
+                    dtstart=start_date
+                )
+                for start_date in exception_dates:
+                    pass
```

Comment:
seem useless

Reply:
This is the best trick to get the last element from the generator into `start_date` variable